tests/test_server.py

```diff
@@ -691,3 +709,136 @@ def count_system_messages_in_recall() -> Tuple[int, List[LettaMessage]]:
     finally:
         # cleanup
         server.delete_agent(user_id, agent_state.id)
+
+
+def test_load_file_to_source(server: SyncServer, user_id: str, agent_id: str, other_agent_id: str, tmp_path):
+    user = server.get_user_or_default(user_id)
+
+    # Create a source
+    source = server.source_manager.create_source(
+        PydanticSource(
+            name="timber_source",
+            embedding_config=EmbeddingConfig.default_config(provider="openai"),
+            created_by_id=user_id,
+        ),
+        actor=user
+    )
+
+    # Create a test file with some content
+    test_file = tmp_path / "test.txt"
+    test_content = "We have a dog called Timber. He likes to sleep."
+    test_file.write_text(test_content)
+
+    # Attach source to agent first
+    agent = server.load_agent(agent_id=agent_id)
+    agent.attach_source(user=user, source_id=source.id, source_manager=server.source_manager, ms=server.ms)
+
+    # Get initial passage count
+    initial_passage_count = server.passage_manager.size(actor=user, agent_id=agent_id, source_id=source.id)
+    assert initial_passage_count == 0
+
+    # Create a job for loading the first file
+    job = server.job_manager.create_job(
+        PydanticJob(
+            user_id=user_id,
+            metadata_={"type": "embedding", "filename": test_file.name, "source_id": source.id},
+        ),
+        actor=user
+    )
+
+    # Load the first file to source
+    server.load_file_to_source(
+        source_id=source.id,
+        file_path=str(test_file),
+        job_id=job.id,
+        actor=user,
+    )
+
+    # Verify job completed successfully
+    job = server.job_manager.get_job_by_id(job_id=job.id, actor=user)
+    assert job.status == "completed"
+    assert job.metadata_["num_passages"]  == 1
+    assert job.metadata_["num_documents"] == 1
+
+    # Verify passages were added
+    first_file_passage_count = server.passage_manager.size(actor=user, agent_id=agent_id, source_id=source.id)
+    assert first_file_passage_count > initial_passage_count
+
+    # Create a second test file with different content
+    test_file2 = tmp_path / "test2.txt"
+    test_content2 = "Timber lives in a house in San Francisco. It's next to the bay. His favorit food is chicken."
+    test_file2.write_text(test_content2)
+
+    # Create a job for loading the second file
+    job2 = server.job_manager.create_job(
+        PydanticJob(
+            user_id=user_id,
+            metadata_={"type": "embedding", "filename": test_file2.name, "source_id": source.id},
+        ),
+        actor=user
+    )
+
+    # Load the second file to source
+    server.load_file_to_source(
+        source_id=source.id,
+        file_path=str(test_file2),
+        job_id=job2.id,
+        actor=user,
+    )
+
+    # Verify second job completed successfully
+    job2 = server.job_manager.get_job_by_id(job_id=job2.id, actor=user)
+    assert job2.status == "completed"
```

Comment:
I think the file you are testing with is very small, so the job is immediately completed - might be better to use a bigger file (I use the `CONTRIBUTING.md` file sometimes, or we could add a bigger test file) to more reliably catch errors